File: Java/ServletServerSide/src/ServerSideSandbox/ApplicationController.java
package ServerSideSandbox;

import java.util.HashMap;

public class ApplicationController {
    // Key String and Value Handler
    private HashMap<String, Handler> handlerMap = new HashMap();
    // Gets a String and a HashMap.
    // In this case the command is the Key
    public void handleRequest(String command, HashMap<String,Object> data){
        // Getting the Value from which the Key is specified from HandlerMap HashMap and I store it in a Handler
        // Method interface
        Handler aCommandHandler = handlerMap.get(command);
        // If this variable is not null then
        if (aCommandHandler != null){
            // Get HandleIt in Speak Handler Class
            aCommandHandler.handleIt(data);
        }
    }

    public void mapCommand(String aCommand, Handler acHandler){
        // Adds the aCommand as the key and Handler object as the value
        handlerMap.put(aCommand,acHandler);
    }
}

File: C#/WeLearnLearningSystem/AngularTestingAndDemo/ApplicationControllers/ApplicationController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Web;

namespace AngularTestingAndDemo.ApplicationControllers
{
    public class ApplicationController
    {
        // Private Instance Variable Which key is a String and Value is a Handler Interface
        private Dictionary<String, WeLearnHandlerInterface> handlerMap = new Dictionary<String, WeLearnHandlerInterface>();

        // Method which receives a String and a HashMap with a String and a Object
        public void handleRequest(String command, Dictionary<String, Object> data)
        {
            try
            {
                // Since Handler is an interface you cannot instantiate it
                // Rather you store the handlerMap HashMap's Key of type String
                // to the Handler
                WeLearnHandlerInterface aCommandHandler = handlerMap[command];
                // If whatever was returned is not null, because its possible
                // Then get Handler's method which parameter is a Dictionary
                // and pass the object to it.
                if (aCommandHandler != null)
                {
                    aCommandHandler.systemHandler(data);
                }
            }
            catch (Exception e)
            {
                e.ToString();
            }
        }


        public void mapCommand(String aCommand, WeLearnHandlerInterface acHandler)
        {

            // Add to the handler HashMap the string and the Handler
            handlerMap.Add(aCommand, acHandler);
        }
    }
}
